dask_ml/model_selection/_search.py

```diff
@@ -399,10 +440,18 @@ def do_fit_and_score(
 
         scores = []
         scores_append = scores.append
+
+        eval_weight_source = _get_weights_source(fit_params)
+        w_train, w_test = None, None
+
         for n in range(n_splits):
-            train_sample_weight, test_sample_weight = _get_fold_sample_weights(
-                sample_weight, cv, n
-            )
+            if eval_weight_source is not None:
+                # format to keys with full information compatible with cv_extract functions
+                keys, vals = _generate_fit_params_key_vals(fit_params, keys_filtered=[eval_weight_source])
+                # dask evaluation requires wrapping into function to allow the function to be evaluated once cv object is resolved
+                def extract_param(cvs,k,v,n,fld):
```

Comment:
move `extract_param` out of the for loop, it does not need to defined multiple times.

Reply:
`cv_extract_params` is slightly different than `extract_param`. `cv_extract_params` is the original Dask function that gets the fit_params information in `{fit_param_key: key_values}` format  for all the `fit_params` . `extract_param` is a one time pull of a single `fit_param` since it needs to be passed on to `score` for a pipeline before it has actually been computed.

Reply:
I mean you can define `extract_param` before for loop.